server/app.js

```diff
@@ -32,16 +36,92 @@ app.get('/products/:id/reviews/', (req, res) => {
     returnObject['reviews'] = results;
     review.find().distinct('grade', (err, grades) => {
       if (err) {
-        res.send(err);
+        res.status(400).send(err);
         return;
       }
       returnObject.grades = grades;
-      res.send(returnObject);
+      res.status(200).send(returnObject);
     });
   }).limit(20);
 });
 
+// adding a POST endpoint to CREATE a new item in the database
+app.post('/products/:id/review', (req, res) => {
+  const productId = req.params.id;
+  const reviewToCreate = {};
+  reviewToCreate.productId = productId;
+
+  if (req.body.grade !== undefined) {
+    reviewToCreate.grade = JSON.parse(req.body.grade);
+  }
+  if (req.body.standards !== undefined) {
+    reviewToCreate.standards = JSON.parse(req.body.standards);
+  }
+  if (req.body.title !== undefined) {
+    reviewToCreate.title = req.body.title;
+  }
+  if (req.body.description !== undefined) {
+    reviewToCreate.description = req.body.description;
+  }
+  if (req.body.rating !== undefined) {
+    reviewToCreate.rating = req.body.rating;
+  }
+  if (req.body.user !== undefined) {
+    reviewToCreate.user = req.body.user;
+  }
+  model.create(reviewToCreate, (err, result) => {
```

Comment:
what if the productID already exists in the database does it over ride it? or should this be an error?

Reply:
So this is the review database. Generally the productId should always exist if a user is trying t add a review. The use case is that the user is on the product page and wants to add a review (so in general the product will always exist).

The problem is that the endpoint right now does not have a way to know if a productId exists are not (it would need to verify this in the products database). Therefore you can theoretically add a review for a productId that does not exist in the product database. 

Does this address your question? 

In summary: 
if a productId exist, the endpoint would add a review for that product and display it when all reviews for that product are fetched. 
if a productId does not exist, the endpoint will still allow for a review to be posted but it will have now respective product in the db database.